[BE][8/N] Remove ShardedTensor from TP FSDP integration test and other tests depending on Sharded Linear

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96254

We removed ShardedLinear in https://github.com/pytorch/pytorch/pull/95948 but it broke TP_FSDP integration test because it is using ShardedTensor in the test. Migrating using DTensor fixes the test. DTensor shards the bias too so that we need to change the test a little bit.

Commit:
[BE][8/N] Remove ShardedTensor from TP FSDP integration test

[ghstack-poisoned]

fduwjj commented:
Looks like the failure about dynamo benchmark is not related to this PR "No CUDA GPUs are available"

Commit:
Update on "[BE][8/N] Remove ShardedTensor from TP FSDP integration test"


We removed ShardedLinear in https://github.com/pytorch/pytorch/pull/95948 but it broke TP_FSDP integration test because it is using ShardedTensor in the test. Migrating using DTensor fixes the test. DTensor shards the bias too so that we need to change the test a little bit.


[ghstack-poisoned]

fduwjj commented on test/distributed/_shard/sharding_plan/test_sharding_plan.py:
```diff
@@ -61,200 +56,6 @@ def build_plan(self, module: nn.Module) -> ShardingPlan:
-    @with_comms(init_rpc=False)
-    @skip_if_lt_x_gpu(TEST_GPU_NUM)
-    @requires_nccl()
-    def test_sharding_plan_simple_megatron(self):
```
These tests are relied on Sharded Linear and we should also remove these two tests.

huydhn commented:
The new periodic multigpu failure https://hud.pytorch.org/pr/96254#11842511229 also look related, so I guess it's another test to be update

Commit:
Update on "[BE][8/N] Remove ShardedTensor from TP FSDP integration test and other tests depending on Sharded Linear"


We removed ShardedLinear in https://github.com/pytorch/pytorch/pull/95948 but it broke TP_FSDP integration test because it is using ShardedTensor in the test. Migrating using DTensor fixes the test. DTensor shards the bias too so that we need to change the test a little bit.


[ghstack-poisoned]

fduwjj commented:
@huydhn Ahhh that's no wonder why I didn't see it. We need to remove that test too. We have one already for DTensor under [test/distributed/](https://github.com/pytorch/pytorch/blob/master/test/distributed/tensor/parallel/test_tp_examples.py). Removed that test too.

huydhn reviewed (approved):
LGTM! You might see buck failure on periodic, but it's broken in trunk at the moment. So the failure is expected.

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot merge -f "failing tests are not related to this PR."

huydhn commented:
@fduwjj I have a small follow-up PR https://github.com/pytorch/pytorch/pull/96431 to cleanup reference to the deleted test.  The multigpu test complains expectedly that it couldn't find the test.
